packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -910,24 +920,24 @@ class _BottomNavigationBarState extends State<BottomNavigationBar> with TickerPr
     return textStyle.fontSize == null ? textStyle.copyWith(fontSize: fontSize) : textStyle;
   }
 
+  // If [IconThemeData] is provided, it should be used.
+  // Otherwise, the [IconThemeData]'s color should be selectedItemColor
+  // or unselectedItemColor.
+  static IconThemeData _effectiveIconTheme(IconThemeData? iconTheme, Color? itemColor) {
+    final IconThemeData effectiveIconTheme = iconTheme ?? const IconThemeData();
+    // Prefer the iconTheme over itemColor and textStyle color if present.
+    // If iconTheme is not set, prefer textStyle color over itemColor
+    return iconTheme == null ? effectiveIconTheme.copyWith(color: itemColor) : effectiveIconTheme;
```

Comment:
Couldn't this just be:

```suggestion
    return iconTheme ?? IconThemeData(color: itemColor);
```

Reply:
I'll make sure this doesn't break the unit tests, cause I remember we were going through that already.